documentation/modules/cruise-control/con-cruise-control-overview.adoc

```diff
@@ -2,54 +2,508 @@
+* Changing topic replication factor
+
+NOTE: Other Cruise Control features are not currently supported, including self healing, notifications, and write-your-own goals.
+
+== Optimization goals
+
+Optimization goals define objectives for rebalancing, such as distributing topic replicas evenly across brokers. 
+
+They are categorized as follows:
+
+* *Main goals* are a list of goals supported by the Cruise Control instance and used in its operations. 
+By default, this list includes all built-in goals. 
+For a goal to be used in other categories, such as default or hard goals, it must first be listed in main goals. 
+To prevent a goal’s usage, remove it from this list.
+* *Hard goals* are preset and must be satisfied for a proposal to succeed.
+* *Soft goals* are objectives that are prioritized during optimization as much as possible, without preventing a proposal from being created if all hard goals are satisfied.
+* *Default goals* refer to the goals used by default when generating proposals. 
+They match the main goals unless specifically set by the user.
+* *Proposal-specific goals* are a subset of default goals configured for specific proposals.
+
+Configure optimization goals in the `Kafka` and `KafkaRebalance` custom resources.
+
+* `Kafka` resource for main, hard, and default goals.
+** Main goals: `Kafka.spec.cruiseControl.config.goals`
+** Hard goals: `Kafka.spec.cruiseControl.config.hard.goals`
+** Default goals: `Kafka.spec.cruiseControl.config.default.goals`
+* `KafkaRebalance` resource for proposal-specific goals.
+** Proposal-specific goals: `KafkaRebalance.spec.goals`
+
+=== Main goals
+
+Main goals are predefined and available to all users in Cruise Control. 
+Goals not listed as main goals cannot be used in Cruise Control operations.
+Some main goals are preset as hard goals.
+
+To simplify configuration, use the inherited main goals unless you need to exclude specific goals from `KafkaRebalance` resources. 
+You can adjust the priority order in the default optimization goals configuration.
+
+Configure main goals in `Kafka.spec.cruiseControl.config.goals`:
+
+* To accept inherited main goals, omit the `goals` property.
+* To modify main goals, specify the goals in descending priority order in the `goals` property.
+
+=== Hard and soft goals
+
+Hard goals are mandatory and must be satisfied for optimization proposals to be generated. 
+Soft goals are best-effort objectives that Cruise Control tries to meet after all hard goals are satisfied.
+The classification of hard and soft goals is fixed in Cruise Control code and cannot be changed.
+
+Cruise Control first prioritizes satisfying hard goals, and then addresses soft goals in the order they are listed.
+A proposal meeting all hard goals is valid, even if it violates some soft goals. 
+
+For example, a soft goal might be to evenly distribute a topic's replicas. 
+Cruise Control will ignore this if it conflicts with hard goals.
```

Comment:
I wouldn't say the Cruise Control will ignore the soft goal, it just continue to generate an optimization proposal even if the soft goal isn't completely satisfied